components/net/http_cache.rs

```diff
@@ -331,8 +333,12 @@ fn create_cached_response(
     // TODO: take must-revalidate into account <https://tools.ietf.org/html/rfc7234#section-5.2.2.1>
     // TODO: if this cache is to be considered shared, take proxy-revalidate into account
     // <https://tools.ietf.org/html/rfc7234#section-5.2.2.7>
-    let has_expired =
-        (adjusted_expires < time_since_validated) || (adjusted_expires == time_since_validated);
+    let has_expired = adjusted_expires <= time_since_validated;
```

Comment:
Also here, we could be returning `None` if `has_expired` is true, and the stored resource doesn't contain a `Date` and `Last-Modified` header, since we need those to do a successful refresh, and if we can't, we should not construct a response that require validation.

Reply:
Mmh, one issue here is that fetch doesn't have a hook to deal with the "could not select a stored response for update", as [Step 7.4](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch) seems to assume that a refresh is always a succesful operation. 

I've filed https://github.com/whatwg/fetch/issues/950